### PR TITLE
Retrieve library hours from LibCal

### LINKED ISSUE #378: Make Homepage Hours pull from libcal

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #378 

Retrieves library hours from LibCal. Because of CORS restrictions that we can't fix (Springshare manages the servers), hours data is pulled in and transformed server side.

Also:

- Adds link to the more hours button
- Organizes javascript with loader classes

Screen Shot 2020-10-02 at 9 20 55 AM

### CODE REVIEW

File: app/javascript/pulfalight/toc_loader.es6 [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import TocBuilder from "../pulfalight/toc.es6"
+
+export default class TocLoader {
+  run() {
+    this.setup_toc()
+    this.setup_before_visit_listenter()
```
Comment (tpendragon) [inline app/javascript/pulfalight/toc_loader.es6:6]: Really small typo - should be `listener`

File: app/services/hours_builder.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  attr_reader :id
+  def initialize(id:)
+    @id = id.to_i
+  end
+
+  def build
+    # Return if id parameter is not an integer
+    return hours_hash(unvailable_message) if id.zero?
+    response = query_lib_cal
```
Comment (tpendragon) [inline app/services/hours_builder.rb:16]: Suggestion: Extract to memo-ized instance method.

File: app/services/hours_builder.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    # Return if LibCal can't be reached
+    return hours_hash(unvailable_message) unless response
+    value = parse_response(response)
```
Comment (tpendragon) [inline app/services/hours_builder.rb:19]: Suggestion: Extract to memo-ized instance method.

File: app/services/hours_builder.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  def self.build(id:)
+    new(id: id).build
+  end
+
+  attr_reader :id
+  def initialize(id:)
+    @id = id.to_i
+  end
+
+  def build
```
Comment (tpendragon) [inline app/services/hours_builder.rb:13]: Suggestion: Extracting some of these methods and making `value` return `hours_hash(unavailable_message)` (or nil I guess) if the id is zero or there's no response will tighten up this logic considerably and make it look nicer and read quicker, I think.

**Review by tpendragon — CHANGES_REQUESTED** [review]
Couple small things.

File: app/services/hours_builder.rb (hunk 4 above)
Comment (eliotjordan) [reply app/services/hours_builder.rb:13]: Refactored.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**